PennMobile/src/main/java/com/pennapps/labs/pennmobile/BookGsrFragment.kt

```diff
@@ -0,0 +1,142 @@
+package com.pennapps.labs.pennmobile
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import butterknife.ButterKnife
+import com.pennapps.labs.pennmobile.api.Labs
+import kotlinx.android.synthetic.main.gsr_details_book.view.*
+import retrofit.Callback
+import retrofit.RetrofitError
+import retrofit.client.Response
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+
+
+class BookGsrFragment : Fragment() {
+
+    // fields for booking
+    internal var firstName: EditText? = null
+    internal var lastName: EditText? = null
+    internal var email: EditText? = null
+    // submit button
+    internal var submit: Button? = null
+
+    private var mLabs: Labs? = null
+
+    // gsr details
+    private var gsrID: String? = null
```

Comment:
It looks like gsrID can be a nonnullable value - all the other gsrID variables (i.e. in the parameters/arguments) here are nonnullable.